Add restaurant details to restaurant-preview of map

JoshuaRamc opened the pull request:
Features Added:
On the map view of the app, when clicking an icon, you can now see restaurant details in the restaurant preview. Formatting and color correction is still in development.
![image](https://user-images.githubusercontent.com/112503024/230747201-7f65e276-d9bf-4467-887f-aca22b41f882.png)

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,80 @@
+          <div style={classes.nameContainer}>{restaurant.name}</div>
+
+          <div>
+            <ReviewStars rating={restaurant.rating}> </ReviewStars>
```
nit: no need for closing tag, just use this instead:
```jsx
<ReviewStars rating={restaurant.rating}/>
```

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,80 @@
+          </div>
+
+          <div>
+            {restaurant.address.address1} {restaurant.address.state}{" "}
```
I imagine that this section is gonna be pretty long, can we store this as a variable like `restaurantAddress`, then render it here?

DLi7077 commented on package-lock.json:
```diff
@@ -0,0 +1,6 @@
+  "lockfileVersion": 3,
+  "requires": true,
+  "packages": {}
+}
```
this file should not be here... prob created by accident

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,80 @@
+    marginLeft: "10px",
+  },
+  nameContainer: {
+    fontSize: "30px",
```
might be too big. I think the largest font used on this website is `1.5rem`, not sure.

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,80 @@
+    height: "100%",
+    width: "100px",
+    marginLeft: "15px",
+    objectFit: "cover",
```
use rem for padding or margin unless you're literally drawing a picture

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,80 @@
+    marginTop: "70px",
+  },
+  exitContainer: {
+    marginLeft: "100px",
```
use rem

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,80 @@
+    marginLeft: "100px",
+  },
+  yelpButtonContainer: {
+    marginTop: "50px",
```
use rem

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,80 @@
+    fontSize: "30px",
+  },
+  categoriesContainer: {
+    marginTop: "70px",
```
use rem

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,80 @@
+          style={classes.imageContainer}
+          src={restaurant.image_url}
+          alt={restaurant.image_url}
+        ></img>
```
no need for closing tag

DLi7077 reviewed (changes requested):
Clear out all the empty lines inside the jsx when you're done

JoshuaRamc commented:
> Features Added: On the map view of the app, when clicking an icon, you can now see restaurant details in the restaurant preview. Formatting and color correction is still in development.

Additionally, I added a link to the restaurant name of the restaurant page. It will navigate to `/restaurantyelp_id=${restaurant.yelp_id}` on click now. Additionally I changed pixels to rem in the appropriate places and formatted the code better.

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,84 @@
+    ", " +
+    restaurant.address.state +
+    ", " +
+    restaurant.address.zip_code;
```
Nit: I see a bunch of `restaurant.address` here. Destructuring the address + state + zip_code from it would prob make it cleaner

```js
const { address1, state, zip } = restaurant.address;
```

DLi7077 replied:
tip: you can destructure and rename like so:
```
const obj = { some_key: 10 };
const { some_key: renamedKey } = obj;
console.log(renamedKey); // 10
```

DLi7077 replied:
also edge case: 
![image](https://user-images.githubusercontent.com/51217000/230526498-92f0e6b0-2d7e-4f12-bb80-e1acdb58aeb8.png)

DLi7077 replied:
here's what the same restaurant looks like on the card:
![image](https://user-images.githubusercontent.com/51217000/230527069-b298420c-df3e-40aa-a3a9-a9e4d5b198f5.png)

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,84 @@
+
+const classes = {
+  previewContainer: {
+    height: "160px",
```
let's reduce this to 100-120px

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,84 @@
+  imageContainer: {
+    height: "100%",
+    width: "100px",
+    marginLeft: "1rem",
```
not a fan of the left margin. If we want spacing on the side, then let's also have spacing around the div. Here's what I mean:
![image](https://user-images.githubusercontent.com/51217000/230526240-f6b34a10-444e-405f-9222-99d24fd35d31.png)

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,84 @@
+    color: "black",
+  },
+  categoriesContainer: {
+    marginTop: "4.8rem",
```
nit: 4.75? :D

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,84 @@
+    marginLeft: "6rem",
+  },
+  yelpButtonContainer: {
+    marginTop: "3.2rem",
```
nit: pref multi of .25 :D

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,84 @@
+        <div>
+          <IconButton style={classes.exitContainer}>
+            <CancelIcon />
+          </IconButton>
```
1. Have it so that the preview closes when the user clicks on this button
2. Use position absolute here because this button should always be on the top right, regardless of the position of other items in this preview card

JoshuaRamc commented:
One thing to note in the most recent commit: the categories position on the restaurant preview is bugged when a restaurant doesn't have a phone number. 
![image](https://user-images.githubusercontent.com/112503024/230747186-4fcc1340-9b23-41ca-b381-8dc8b7b8f61e.png)
![image](https://user-images.githubusercontent.com/112503024/230747189-5e753d65-be1b-41c7-9fff-4d81c0160c99.png)

DLi7077 commented:
@JoshuaRamc 
That can prob be fixed by having marginTop auto on the categories. That way, it'll be as far away from the top as possible.

Also not a fan of bolded text

JoshuaRamc commented:
> @JoshuaRamc That can prob be fixed by having marginTop auto instead of 1rem on the categories. That way, it'll be as far away from the top as possible.
> 
> Also not a fan of bolded text

I had tried marginTop with auto before. Did not solve the problem. And noted, will remove the bolded text.

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,103 @@
+  },
+};
+
+function formatPhoneNumber(phoneNumberString) {
```
Add a source if you found this online. Otherwise if you did this yourself - holy shit nice

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,103 @@
+function RestaurantPreview({ restaurant, hideRestaurant }) {
+  const restaurantPage = "/restaurant?yelp_id=" + restaurant.yelp_id;
+  const phoneNumber = formatPhoneNumber(restaurant.phone);
+  const streetAddress = restaurant.address.address1 ?? "";
+  const restaurantAddress =
+    streetAddress +
+    " " +
+    restaurant.address.state +
+    ", " +
+    restaurant.address.zip_code;
```
I think this would look cleaner:
```js
const { address1, state, zip_code } = restaurant.address;
const restaurantAddress = `${address1} ${state}, ${zip_code}`;
```

Optional: If you want to rename `address1` to `streetAddress` , you can rename it in the destructure:
```js
const { address1: streetAddress, state, zip_code } = restaurant.address;
const restaurantAddress = `${streetAddress} ${state}, ${zip_code}`;
```

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,103 @@
-    <div style={{ height: "160px", width: "100%", backgroundColor: "gray" }}>
-      {restaurant.name}
+    <div style={classes.previewContainer}>
+      <div style={{ display: "flex" }}>
```
Not sure if we need this

DLi7077 commented on client/src/pages/Search/MapView/index.js:
```diff
@@ -11,6 +11,11 @@ export default function MapView({ longitude, latitude, rows, updateFields }) {
   }
 
+  function handleHideRestaurant(restaurant) {
+    setSelectedRestaurant(restaurant);
```
Do we need to set the restaurant if we're hiding the component?

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,103 @@
+    justifyContent: "space-between",
+  },
+  imageContainer: {
+    height: "75%",
```
try height 100%

JoshuaRamc replied:
does not work, lol. looks jank with 100%.

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,103 @@
+    objectFit: "cover",
+  },
+  detailsContainer: {
+    marginLeft: "0.25rem",
```
so this is the container that shows the restaurant info...
We can fix the category position bug two ways:
1. Easiest: slap a `position absolute` on it relative to the bottom of this container, and make this container `height 100%`
2. Flexbox: slap `display flex` and `flexDirection column` so it aligns downwards
then after that, `marginTop auto` on the categories div will probably fix it.

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,103 @@
+};
+
+function formatPhoneNumber(phoneNumberString) {
+  var cleaned = ("" + phoneNumberString).replace(/\D/g, "");
+  var match = cleaned.match(/^(1|)?(\d{3})(\d{3})(\d{4})$/);
+  if (match) {
+    var intlCode = match[1] ? "+1 " : "";
+    return [intlCode, "(", match[2], ") ", match[3], "-", match[4]].join("");
+  }
```
nit: use `const` I think most ppl never use `var`, always `let` or `const` so let's just stick with that

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,103 @@
+          </div>
+
+          <div>
+            <div>{restaurantAddress}</div>
```
can we give this the orange tint just like in restaurant card?

JoshuaRamc replied:
give what an orange tint? the preview?

DLi7077 commented:
@JoshuaRamc 
I think the restaurant details should also be in the padded region of the div
![image](https://user-images.githubusercontent.com/51217000/230747878-39c7a2d4-2359-4321-8010-8019c438e645.png)

the light green region (outside of the purple dashes) is the padding (like 2px). Ideally that padding should surround the entire content, not just on the image

Reduce the padding by maybe 50% and have the content be in the same container as the image. The overflowing title doesn't feel right imo
![image](https://user-images.githubusercontent.com/51217000/230747929-7fa24f93-87be-4beb-85a5-e8d0fa94b628.png)

JoshuaRamc commented:
![image](https://user-images.githubusercontent.com/112503024/230750719-429e4080-5bb0-4d97-bb9e-54e4549b79e7.png)
looking nice!

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,111 @@
+    color: "rgb(80,80,80)",
+  },
+  exitContainer: {
+    postion: "absolute",
```
lmao "postion"

JoshuaRamc replied:
lol

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,111 @@
+};
+
+function formatPhoneNumber(phoneNumberString) {
+  // Resource: https://stackoverflow.com/questions/8358084/regular-expression-to-reformat-a-us-phone-number-in-javascript
```
move comments above the function unless it's related to the line right after

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,111 @@
+
+function formatPhoneNumber(phoneNumberString) {
+  // Resource: https://stackoverflow.com/questions/8358084/regular-expression-to-reformat-a-us-phone-number-in-javascript
+  const cleaned = ("" + phoneNumberString).replace(/\D/g, "");
```
why do we need an empty string here?

JoshuaRamc replied:
unsure honestly, not my algo :( will fix it up tho

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,111 @@
+    const intlCode = match[1] ? "+1 " : "";
+    return [intlCode, "(", match[2], ") ", match[3], "-", match[4]].join("");
+  }
+  return null;
```
nit: new line before return

DLi7077 commented on client/src/pages/Search/MapView/RestaurantPreview.js:
```diff
@@ -1,7 +1,111 @@
+          </div>
+          <div>
+            <div style={{ color: "rgb(80,80,80)" }}>{phoneNumber}</div>
+          </div>
```
redundant div here I think

DLi7077 commented on client/src/pages/Search/MapView/index.js:
```diff
@@ -11,6 +11,10 @@ export default function MapView({ longitude, latitude, rows, updateFields }) {
     setShowing(true);
   }
 
+  function handleHideRestaurant(restaurant) {
```
no need for restaurant param here

DLi7077 reviewed (changes requested):
few more changes and we're good to go

DLi7077 reviewed (approved):
Good work!
